File: s-recursos/src/main/java/co/edu/javeriana/arqui/rest/model/Clinica.java
package co.edu.javeriana.arqui.rest.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;

@Entity
public class Clinica {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nombre;
    private String direccion;
    private String telefono;

    public Clinica() {
    }

    public Clinica(String nombre, String direccion, String telefono) {
        this.nombre = nombre;
        this.direccion = direccion;
        this.telefono = telefono;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getDireccion() {
        return direccion;
    }

    public void setDireccion(String direccion) {
        this.direccion = direccion;
    }

    public String getTelefono() {
        return telefono;
    }

    public void setTelefono(String telefono) {
        this.telefono = telefono;
    }
}

File: ms-clinicas/src/main/kotlin/co/edu/javeriana/arqui/model/Clinica.kt
package co.edu.javeriana.arqui.model

import jakarta.persistence.*
import jakarta.validation.constraints.NotBlank

@Entity
@Table(name = "clinicas")
open class Clinica {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    open var id: Long? = null

    @field:NotBlank
    @Column(nullable = false)
    open var nombre: String? = null

    @field:NotBlank
    @Column(nullable = false)
    open var direccion: String? = null

    @field:NotBlank
    @Column(nullable = false)
    open var telefono: String? = null

    // Constructor sin argumentos (necesario para Hibernate)
    constructor()

    // Constructor conveniente
    constructor(nombre: String, direccion: String, telefono: String) {
        this.nombre = nombre
        this.direccion = direccion
        this.telefono = telefono
    }
}
